### PR TITLE
[Dubbo-7038]Returns the same reference if no filtering has occurred, to save memory.

### PR DESCRIPTION
**Pull request by HeYuqiu**
## What is the purpose of the change

[7038]https://github.com/apache/dubbo/issues/7038

### CODE REVIEW

**Comment by kimmking** [comment]
I don't think keep original reference is a goog way.

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/UrlUtils.java [hunk 1]
```diff
@@ -470,6 +470,16 @@ public static boolean isServiceKeyMatch(URL pattern, URL value) {
...
     public static List<URL> classifyUrls(List<URL> urls, Predicate<URL> predicate) {
+        boolean filter = false;
+        for (URL url : urls) {
+            if (!predicate.test(url)) {
+                filter = true;
+                break;
+            }
+        }
+        if (!filter) {
+            return urls;
```
Comment (kimmking) [inline dubbo-common/src/main/java/org/apache/dubbo/common/utils/UrlUtils.java:481]: keep an outer variable is dangerous.

**Review by kimmking — CHANGES_REQUESTED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/common/utils/UrlUtils.java (hunk 1 above)
Comment (chickenlj) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/UrlUtils.java:481]: Better wrapping `urls` with unmodifiable?

Comment (chickenlj) [reply dubbo-common/src/main/java/org/apache/dubbo/common/utils/UrlUtils.java:481]: Let's keep this snippet as-is for this is not a critical path for performance improvement.

**Pull request closed without merging**